Monika-After-Story/MonikaModDev
# Some more NOU improvements

### Changes:
- Allow to ask to change house rules after the game is over
- Delayed desk sprite init to init 500, also guardrailed global map access with a try/except
- Fixed an issue where Monika wouldn't start a minigame she picked in the `mas_mood_bored` event
- Implemented a new house rules thanks to @Legendkiller21 
- - Allows to reflect a +2 or +4 with a +4, and reflect skips using a skip of any colour


### Testing:
- Test the new way of changing house rules
- - After Monika wins a game with points
- - After player wins a game with points
- - After Monika wins a game without points
- - After player wins a game without points
- - - For all of the above, test both cases where you continue to play AND leave after changing the rules
- - - You can use `renpy.get_return_stack()` to verify we're not adding an extra call stack
- Verify desk sprites still work as before
- Verify Monika can start any game from `mas_mood_bored` and the games have correct names (NOU should be capitalised, the rest not)
- Verify the new house rule works and Monika still plays good with and without it
